### PR TITLE
allow transactions with 0 witnesses to run

### PR DESCRIPTION
**Pull request by mkv-vcm**

### CODE REVIEW

File: src/signTx.c [hunk 1]
```diff
@@ -199,7 +199,9 @@ static inline void advanceStage()
 		ctx->stage = SIGN_STAGE_WITNESSES;
 		initTxWitnessCtx();
 
-		break;
+		if (ctx->numWitnesses != 0) {
+			break;
+		}
```
Comment (janmazak) [inline src/signTx.c:204]: The comment "// intentional fallthrough" applies here.

File: src/signTx.c [hunk 2]
```diff
@@ -199,7 +199,9 @@ static inline void advanceStage()
 		ctx->stage = SIGN_STAGE_WITNESSES;
 		initTxWitnessCtx();
 
-		break;
+		if (ctx->numWitnesses != 0) {
```
Comment (janmazak) [inline src/signTx.c:202]:
```suggestion
		if (ctx->numWitnesses > 0) {
```

File: src/signTx.c (hunk 1 above)
Comment (mkv-vcm) [reply src/signTx.c:204]: True.

File: src/signTx.c (hunk 2 above)
Comment (mkv-vcm) [reply src/signTx.c]: Not sure why it's better, but if you prefer that, can be done.

Comment (janmazak) [reply src/signTx.c]: Mostly just because we do it like this in other places. (probably because human are less good at processing negations?...)

**Comment by janmazak** [comment]
Will be merged directly.

**Pull request closed without merging**